tapir/coop/forms.py

```diff
@@ -247,3 +251,154 @@ class Meta:
+            self.instance.pay_out_day = cancellation_date
+            self.instance.paid_out = True
+
+    def validate_share_owner(self):
+        share_owner = self.cleaned_data.get("share_owner")
+        if self.instance.pk is None:
+            for already_resigned_member in MembershipResignation.objects.values(
+                "share_owner"
+            ):
+                if already_resigned_member["share_owner"] == share_owner.id:
+                    self.add_error(
+                        "share_owner",
+                        ValidationError(_("This member is already resigned.")),
+                    )
+                    break
+        return share_owner
+
+    def validate_choices(self):
+        coop_buys_shares_back = self.cleaned_data.get("coop_buys_shares_back")
+        willing_to_gift_shares_to_coop = self.cleaned_data.get(
+            "willing_to_gift_shares_to_coop"
+        )
+        transfering_shares_to = self.cleaned_data.get("transfering_shares_to")
+        only_one_choice_allowed_error_message = _("Please take only one choice.")
+
+        if coop_buys_shares_back and willing_to_gift_shares_to_coop:
+            self.add_error(
+                "coop_buys_shares_back", only_one_choice_allowed_error_message
+            )
+            self.add_error(
+                "willing_to_gift_shares_to_coop", only_one_choice_allowed_error_message
+            )
+        elif transfering_shares_to != None and (
+            coop_buys_shares_back or willing_to_gift_shares_to_coop
+        ):
+            self.add_error(
+                "transfering_shares_to", only_one_choice_allowed_error_message
+            )
+            if coop_buys_shares_back:
+                self.add_error(
+                    "coop_buys_shares_back", only_one_choice_allowed_error_message
+                )
+            elif willing_to_gift_shares_to_coop:
+                self.add_error(
+                    "willing_to_gift_shares_to_coop",
+                    only_one_choice_allowed_error_message,
+                )
+
+        if (
+            transfering_shares_to == None
+            and not willing_to_gift_shares_to_coop
+            and not coop_buys_shares_back
+        ):
+            make_at_least_one_choice_error_message = "Please make a least one choice."
+            self.add_error(
+                "transfering_shares_to",
+                ValidationError(_(make_at_least_one_choice_error_message)),
+            )
+            self.add_error(
+                "willing_to_gift_shares_to_coop",
+                ValidationError(_(make_at_least_one_choice_error_message)),
+            )
+            self.add_error(
+                "coop_buys_shares_back",
+                ValidationError(_(make_at_least_one_choice_error_message)),
+            )
+        return (
+            coop_buys_shares_back,
+            willing_to_gift_shares_to_coop,
+            transfering_shares_to,
+        )
```

Comment:
That's for the clean_* functions that Django will call automatically, but for the validate_* functions you wrote it's not needed since Django won't call them.